drivers/gpio/gpio_renesas_rz.c

```diff
@@ -265,150 +362,143 @@ static int gpio_rz_port_clear_bits_raw(const struct device *dev, gpio_port_pins_
 static int gpio_rz_port_toggle_bits(const struct device *dev, gpio_port_pins_t pins)
 {
 	const struct gpio_rz_config *config = dev->config;
-	struct gpio_rz_data *data = dev->data;
-	bsp_io_port_pin_t port_pin;
-	struct gpio_rz_flags rz_flags;
-	ioport_size_t value = 0;
-	fsp_err_t err;
+	uint8_t val;
+	uint8_t port = PORT(config->fsp_port);
 
-	for (uint8_t idx = 0; idx < config->ngpios; idx++) {
-		if (pins & (1U << idx)) {
-			port_pin = config->fsp_port | idx;
-			gpio_rz_pin_config_get_raw(port_pin, &rz_flags);
-			if (rz_flags.gpio_flags & GPIO_OUTPUT_INIT_HIGH) {
-				value &= (1U << idx);
-			} else if (rz_flags.gpio_flags & GPIO_OUTPUT_INIT_LOW) {
-				value |= (1U << idx);
-			}
-		}
-	}
-	err = config->fsp_api->portWrite(data->fsp_ctrl, config->fsp_port, value,
-					 (ioport_size_t)pins);
-	if (err != FSP_SUCCESS) {
-		return -EIO;
-	}
+	IF_ENABLED(CONFIG_GPIO_RENESAS_RZ_TYPE1, (
+		val = REG_P_READ(regs.base, port);
+		REG_P_WRITE(regs.base, port, val ^ pins);
+	))
+
+	IF_ENABLED(CONFIG_GPIO_RENESAS_RZ_TYPE2, (
+		uint8_t rselp = REG_RSELP_READ(regs.rselp, port);
+
+		uint8_t pins_ns = rselp & pins;
+		uint8_t pins_s = ~rselp & pins;
+
+		val = REG_P_READ(regs.base.ns, port);
+		REG_P_WRITE(regs.base.ns, port, val ^ pins_ns);
+
+		val = REG_P_READ(regs.base.s, port);
+		REG_P_WRITE(regs.base.s, port, val ^ pins_s);
+	))
```

Comment:
Hi @nhutnguyenkc, 
Could you please review the issues in the Quality Gate and address them? I’ve included these changes for your reference, feel free to share your thoughts or suggest alternatives.

```suggestion
#if defined(CONFIG_GPIO_RENESAS_RZ_TYPE1) && (CONFIG_GPIO_RENESAS_RZ_TYPE1 == 1)
	pincfg->mode = gpio_rz_get_mode(regs.base, port, pin);
	pincfg->func = gpio_rz_get_func(regs.base, port, pin);
	pincfg->out_state = gpio_rz_get_out_state(regs.base, port, pin);
#else
	pincfg->mode = gpio_rz_get_mode(regs.base.ns, port, pin);
	pincfg->mode |= gpio_rz_get_mode(regs.base.s, port, pin);

	pincfg->func = gpio_rz_get_func(regs.base.ns, port, pin);
	pincfg->func |= gpio_rz_get_func(regs.base.s, port, pin);

	pincfg->out_state = gpio_rz_get_out_state(regs.base.ns, port, pin);
	pincfg->out_state |= gpio_rz_get_out_state(regs.base.s, port, pin);
#endif
```